make_folder/build.sh

```diff
@@ -0,0 +1,301 @@
+            		cat $partfile   >>$bodyfile
+	    else	# add custom part as extra part
+			# set booktitle for custompart
+			if [ $(echo $CHAPTERS $APPENDICES | wc -w ) -gt 1 ]
+			then	BOOKTITLE="Appendices"
+			else	BOOKTITLE="Appendix"
+			fi
+			echod "Adding the custom part at the end."
+			fill_part $partfile
+	    fi
+    fi
+    }
+
+build_xml() {
+	echo -n "Parsing config $BOOKSDIR/$book/config ... "
+	CHAPTERS=""
+	APPENDICES=""
+	. $BOOKSDIR/$book/config
+	. $BOOKSDIR/$book/version
+
+	echod "This book contains:"
+	echod "MINIBOOKS = $MINIBOOKS"
+	echod "CHAPTERS = $CHAPTERS"
+	echod "APPENDICES = $APPENDICES"
+
+	VERSIONSTRING=lt-$MAJOR.$MINOR
+
+	echo "generating book $book (titled \"$BOOKTITLE\")"
+	[ -d $OUTPUTDIR ] || mkdir $OUTPUTDIR
+
+	BOOKTITLE2=$(echo $BOOKTITLE | sed -e 's/\ /\_/g' -e 's@/@-@g' )
+	filename=$BOOKTITLE2-$VERSIONSTRING-$DATECODE
+	xmlfile=$OUTPUTDIR/$filename.xml
+	tmp_xmlfile=$OUTPUTDIR/$filename.tmp.xml
+	pdffile=$OUTPUTDIR/$filename.pdf
+	headerfile=$OUTPUTDIR/section_header.xml
+	footerfile=$OUTPUTDIR/section_footer.xml
+	bodyfile=$OUTPUTDIR/section_body.xml
+	partfile=$OUTPUTDIR/part.xml
+	partcustomfile=$OUTPUTDIR/partcustom.xml
+
+	# make header
+	build_header
+
+	# make footer
+	build_footer
+
+	# make body
+	build_body
+
+	# build master xml
+	echo "Building $xmlfile"
+	cat $headerfile  > $xmlfile
+	cat $bodyfile   >> $xmlfile
+	cat $footerfile >> $xmlfile
+	}
+
+build_pdf() {
+	#set -x
+	validate_dependencies
+	set_xsl
+	set_JAVA
+	echo 
+	echo "---------------------------------"
+	echo "Generating $pdffile"
+	tail -n +2 $xmlfile > $tmp_xmlfile
+	eval $(echo fop -xml $tmp_xmlfile -xsl $XSLFILE -pdf $pdffile ) >&2
+	#eval $(echo fop -xml $tmp_xmlfile -xsl $XSLFILE -pdf $pdffile $EXECDEBUG) >&2 #there is an issue with EXECDEBUG param
+	#fop -xml $tmp_xmlfile -xsl $XSLFILE -pdf $pdffile 
+	ln -s $V $filename.xml $OUTPUTDIR/book.pdf
+	#set +x
+	echo "---------------------------------"
+	}
+
+build_html() {
+    [ -d $HTMLDIR ] && rm -rf $V $HTMLDIR
+    mkdir $V $HTMLDIR || ( echor Error creating $HTMLDIR; exit 1 )
+    mkdir $V $HTMLIMGDIR || ( echor Error creating $HTMLIMGDIR; exit 1 )
+
+    # We only need the one xml file
+    cp $V $xmlfile $HTMLDIR || ( echor error copying $xmlfile ; exit 1 )
+
+    # Locate the used images in the xml file
+    images=`grep imagedata $HTMLDIR/*.xml | cut -d/ -f2 | cut -d\" -f1`
+
+    # Copy all the used images
+    for img in $images
+    do
+         echo Copying $img to $HTMLIMGDIR ...
+         cp $V "$IMAGESDIR/$img" $HTMLIMGDIR/ || echor Error copying $img 
+    done
+
+    # Copy css file to html directory
+    cp $HTMLCSS $HTMLDIR
+    # Run xmlto in $HTMLDIR to generate the html
+    echo "Converting xml to html ..."
+    ( cd $HTMLDIR && xmlto html *.xml --skip-validation -m ../../$HTMLXSL 2>&1 | grep -v "Writing" ) || ( echor  Error generating the html $HTMLDIR ; exit 1 )
+
+    # don't need the xml anymore in the $HTMLDIR
+    rm $HTMLDIR/*.xml
+
+}
```

Comment:
This file is imported from the `build` repository I presume? Good idea, I can archive that one then.

Reply:
yes and not - I still have not finished it and have not tested, so leave it be for now.